### ISSUE TITLE
Flickering redraw (GVim)

### ISSUE THREAD

**Issue by AlexMasterov** [vim/vim#1637]
Use `redraw` in [this code](https://github.com/Shougo/denite.nvim/blob/737f7b171966b809e4a8da2dcb2b17c83985ea1e/rplugin/python3/denite/prompt/prompt.py#L149) is causing flickering.

It is reproduced on GVim in Windows and Linux.
It is not reproduced on terminal Vim.
(It is not reproduced on NVim).

[It is Vim problem](https://github.com/Shougo/denite.nvim/issues/249#issuecomment-294123151)? (won't fix the issue)
## Environment Information
**OS:** Windows 10 (15063)
**Vim:** 8.0.556 (compiled by `gcc` 6.3)

## The reproduce ways
1. Install [denite.nvim](https://github.com/Shougo/denite.nvim)
2. `:Denite line -mode=insert`

## Screen shot
![Imgur](http://i.imgur.com/74SMmDw.gif)

**Comment by nuko8** [comment]
> It is reproduced on GVim in Windows and Linux.

Did you make sure that it happened on Linux yourself?  Which GUI?  As we have various GUIs for Vim running on Linux, it would help us much to know where to start if we know the GUI.

> (It is not reproduced on NVim).

Did you check that with any of nvim GUIs?  Although I don't know any of them,  they always offer a list of them.  So you could examine the issue with a few or all of them before having come to that parenthesized conclusion.

> It is Vim problem? (won't fix the issue)

I'm not sure for now.  The following are what came to my mind immediately: 

1.  Redraw is done differently in Windows and Linux using different GUI toolkits.  So if the both show a similar flicker in common,  the cause of it could lie in the core part of Vim.  This hypothesis, however, is turned out to be difficult to explain why the terminal Vim doesn't show the flicker.  Then, "Does the cause really lie in the core part of Vim?" is a natural question.

2.  For the Windows and Linux cases, the common participants are the Vim core and the mentioned plugin.  As for those two different GUIs, they are doing their job passively following the instructions from the both.  So the cause could lie in either the Vim core or the plugin.

3.  According to the code snippet to which the "this code" link points, one can see a special handling for MacVim, another GUI using another GUI toolkit.   Then, why does the author of the plugin do that for MacVim, but not for Vim?

4.  Probably, the most decisive way to answer the question "It is Vim problem? (won't fix the issue)" is to provide a reproduce procedure without using the mentioned plugin.  Will you do that?

**Comment by AlexMasterov** [comment]
> Did you make sure that it happened on Linux yourself? Which GUI?

I myself can't reproduce on Ubuntu 17.04 (VM) with compiled GUI `gtk2` and `gnome2`.

> Did you check that with any of nvim GUIs?

I checked only with Neovim QT, which is now the default for Windows; this problem is not reproduced.

> provide a reproduce procedure without using the mentioned plugin. Will you do that?

While I can't write a clean test case.

On Windows with GVim I use:
```vim
set renderoptions=type:directx,gamma:2.2,contrast:0.5,level:0.5,geom:1,renmode:3,taamode:2
```

But if `renderoptions` option is not set (default is empty) then the text in the prompt almost no flickering. 
The flickering cursor is not affected. Flicker workflow.

**Comment by chrisbra** [comment]
closing according to [comment from Shougo](https://github.com/Shougo/denite.nvim/issues/249#issuecomment-294644828)

**Comment by AlexMasterov** [comment]
[Fix from Shougo](https://github.com/Shougo/denite.nvim/commit/de58a9b854fbd8e9f3fb304bf20f44999309ac68) just hides the cursor, but does not fix the flickering on Windows 10 in GVim.

@chrisbra So, nothing has changed. Issue not resolved.

**Comment by Shougo** [comment]
> Fix from Shougo just hides the cursor, but does not fix the flickering on Windows 10 in GVim.

I have fixed the problem(by guicursor hack) in GVim Linux, but it is not fixed in GVim Windows...

**Comment by vim-ml** [comment]
Hi there,

2017-4-21(Fri) 15:00:13 UTC+9 Christian Brabandt:
> Reopened #1637.

Confirm an attached patch.

--
Best regards,
Hirohito Higashi (h_east)

**Comment by AlexMasterov** [comment]
Solved by @Shougo via [e5b58f0](https://github.com/Shougo/denite.nvim/commit/e5b58f005ea8216744953f2c20db3e03981bba94) :relaxed:

**Comment by vim-ml** [comment]
Hi Alex,

2017-4-25(Tue) 17:58:14 UTC+9 Alex Masterov:
> Closed #1637.
> 
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub

It does not seem that this commit relates to this Issue.
This commit modifies the flickering of redrawstatus. But, This issue has a problem with flickering of GUI cursor.

Have you truly solved this issue problem with this commit?

--
Best regards,
Hirohito Higashi (h_east)
